### ISSUE TITLE
apply xl-value-format overwrites the color of total row

### ISSUE THREAD

**Issue by msgoussi** [cbailiss/pivottabler#46]
library(pivottabler)
pt <- PivotTable$new()
pt$addData(bhmtrains)
pt$addColumnDataGroups("TrainCategory")
cgrps <- pt$addColumnDataGroups("PowerType")
colorText <- function(grp) {
  if(isTRUE(grp$values=="DMU")) grp$setStyling(list(color="blue"))
  else if(isTRUE(grp$values=="EMU")) grp$setStyling(list(color="green"))
  else if(isTRUE(grp$values=="HST")) grp$setStyling(list(color="red"))
}
invisible(lapply(cgrps, colorText))
pt$addRowDataGroups("TOC",  atLevel=1, addTotal=FALSE)
pt$addRowGroup(caption="Total", isOutline=TRUE, isTotal=TRUE, isLevelTotal=TRUE,
               styleDeclarations=list(color="blue"), cellStyleDeclarations=list(color="blue"))
pt$defineCalculation(calculationName="TotalTrains", summariseExpression="n()",
                     format = "%.0f",
                     cellStyleDeclarations = list("xl-value-format" = "#,##0"))
pt$renderPivot()

**Comment by cbailiss** [comment]
Thanks, this looks like a bug.  There is an intersection of style declarations (headers, calculation and cell-specific) that isn't being properly handled.

One workaround is to reapply the colour to the total:

```
library(pivottabler)
pt <- PivotTable$new()
pt$addData(bhmtrains)
pt$addColumnDataGroups("TrainCategory")
cgrps <- pt$addColumnDataGroups("PowerType")
colorText <- function(grp) {
  if(isTRUE(grp$values=="DMU")) grp$setStyling(list(color="blue"))
  else if(isTRUE(grp$values=="EMU")) grp$setStyling(list(color="green"))
  else if(isTRUE(grp$values=="HST")) grp$setStyling(list(color="red"))
}
invisible(lapply(cgrps, colorText))
pt$addRowDataGroups("TOC", atLevel=1, addTotal=FALSE)
totalGroup <- pt$addRowGroup(caption="Total", isOutline=TRUE, isTotal=TRUE, isLevelTotal=TRUE,
               styleDeclarations=list(color="blue"), cellStyleDeclarations=list(color="blue"))
pt$defineCalculation(calculationName="TotalTrains", summariseExpression="n()",
                     format = "%.0f",
                     cellStyleDeclarations = list("xl-value-format" = "#,##0"))
pt$evaluatePivot()
rowNumber <- pt$findGroupRowNumbers(totalGroup)
pt$setStyling(rFrom=rowNumber, rTo=rowNumber, cFrom=1, cTo=pt$columnCount, declarations=list(color="blue"))
pt$renderPivot()
```

**Comment by msgoussi** [comment]
if you have many total group created via sapply/lapply, i think findCells will be a better option. Can i find row number with specific value!!!! 

Also in regards of styling, what do you think of changing the rFrom and rTo arguments to one arguments, in case if you have not sequential rows  ie c(1, 5, 10). just a thought.

**Comment by cbailiss** [comment]
A fix has been implemented for this on the development branch, which will likely be released in a few weeks.

Good suggestions re: vectorising and non-sequential groups.  #47 has been updated and #50 plus #51 added.  These should be implemented and released at the same time.

**Comment by cbailiss** [comment]
This fix is now on CRAN in v1.5.0.
